src/api/create-vis.js

```diff
@@ -1,34 +1,79 @@
 var _ = require('underscore');
 var Vis = require('../vis/vis');
+var Loader = require('../core/loader');
+var Promise = require('./promise');
 
-var createVis = function(el, vizjson, options, callback) {
+var createVis = function (el, vizjson, options, callback) {
   if (!el) {
-    throw new TypeError("a DOM element should be provided");
+    throw new TypeError('a DOM element must be provided');
+  }
+  if (!vizjson) {
+    throw new TypeError('a vizjson URL or object must be provided');
+  }
+
+  if (typeof vizjson === 'string') {
+    return loadVizJSONAndCreateVis(el, vizjson, options, callback);
   }
 
-  var
-  args = arguments,
-  fn   = args[args.length -1];
+  var args = arguments;
+  var fn = args[args.length - 1];
 
   if (_.isFunction(fn)) {
     callback = fn;
   }
+  if (typeof el === 'string') {
+    el = document.getElementById(el);
+  }
+
+  var DEFAULT_OPTIONS = {
```

Comment:
Considering this is a intended to be constant, why not move it outside the function?